### PR TITLE
Moved getCreatureCondition to metatable

### PR DESCRIPTION
**Pull request by raymondtfr**
Renamed function name ``getCreatureCondition`` to ``hasCondition`` and parameter **condition** to **type** as it stands for the condition type.

### CODE REVIEW

**Comment by ranisalt** [comment]
You should group this function code with the others (addCondition, removeCondition)

File: data/lib/compat/compat.lua [hunk 1]
```diff
@@ -181,6 +181,7 @@ function doSetCreatureDropLoot(cid, doDrop) local c = Creature(cid) return c ~=
 function doChangeSpeed(cid, delta) local c = Creature(cid) return c ~= nil and c:changeSpeed(delta) or false end
 function doAddCondition(cid, conditionId) local c = Creature(cid) return c ~= nil and c:addCondition(conditionId) or false end
 function doRemoveCondition(cid, conditionType, subId) local c = Creature(cid) return c ~= nil and (c:removeCondition(conditionType, CONDITIONID_COMBAT, subId) or c:removeCondition(conditionType, CONDITIONID_DEFAULT, subId) or true) end
+function getCreatureCondition(cid, type, subId) local c = Creature(cid) return c ~= nil and c:hasCondition(type, subId) or false end
```
Comment (ranisalt) [inline data/lib/compat/compat.lua]: For simplicity, I suggest you use:

```
function getCreatureCondition(cid, ...) local c = Creature(cid) return c ~= nil and c:hasCondition(...) or false end
```

File: src/luascript.cpp [hunk 2]
```diff
@@ -7253,6 +7236,21 @@ int LuaScriptInterface::luaCreatureRemoveCondition(lua_State* L)
...
+int LuaScriptInterface::luaCreatureHasCondition(lua_State* L)
+{
+	//creature:hasCondition(conditionType, subId)
+	Creature* creature = getUserdata<Creature>(L, 1);
+	if (!creature) {
+		lua_pushnil(L);
+		return 1;
+	}
+
+	ConditionType_t condition = getNumber<ConditionType_t>(L, 2);
```
Comment (raymondtfr) [inline src/luascript.cpp]: ``conditionType``

File: src/luascript.cpp [hunk 3]
```diff
@@ -7253,6 +7236,21 @@ int LuaScriptInterface::luaCreatureRemoveCondition(lua_State* L)
...
+	uint32_t subId = getNumber<uint32_t>(L, 3, 0);
+	pushBoolean(L, creature->hasCondition(condition, subId));
```
Comment (raymondtfr) [inline src/luascript.cpp]: ``conditionType``

File: src/luascript.cpp (hunk 2 above)
Comment (ranisalt) [inline src/luascript.cpp]: It's because other functions call this parameter `condition`. You can create another PR renaming once this is merged.

**Comment by ranisalt** [comment]
I merged with master because the rebase was very strange and it really doesn't matter now. Is it ok @raymondftr?

**Pull request closed without merging**